src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1132,13 +1132,14 @@ internal virtual void IntersectWithEnumerable(IEnumerable<T> other)
                 if (Contains(item))
                 {
                     toSave.Add(item);
+                    DoRemove(item);
                 }
             }
 
-            if (toSave.Count < Count)
+            Clear();
+            foreach (T item in toSave)
             {
-                Clear();
-                AddAllElements(toSave);
+                AddIfNotPresent(item);
```

Comment:
Does changing AddAllElements like that break UnionWith?  Rather than adding more code (even if just a line or two), it'd be great if we could just reuse the existing helpers already there, making them better in the process.